packages/core/src/amazonq/lsp/lspController.ts

```diff
@@ -340,31 +351,35 @@ export class LspController {
                     amazonqIndexMemoryUsageInMB: usage ? usage.memoryUsage / (1024 * 1024) : undefined,
                     amazonqIndexCpuUsagePercentage: usage ? usage.cpuUsage : undefined,
                     amazonqIndexFileSizeInMB: totalSizeBytes / (1024 * 1024),
-                    credentialStartUrl: AuthUtil.instance.startUrl,
+                    credentialStartUrl: buildIndexConfig.startUrl,
                 })
             } else {
-                getLogger().error(`LspController: Failed to build index of project`)
-                telemetry.amazonq_indexWorkspace.emit({
-                    duration: performance.now() - start,
-                    result: 'Failed',
-                    amazonqIndexFileCount: 0,
-                    amazonqIndexFileSizeInMB: 0,
-                })
+                // TODO: Re-enable this code path for LSP 0.1.20+
+                // getLogger().error(`LspController: Failed to build index of project`)
+                // telemetry.amazonq_indexWorkspace.emit({
+                //     duration: performance.now() - start,
+                //     result: 'Failed',
+                //     amazonqIndexFileCount: 0,
+                //     amazonqIndexFileSizeInMB: 0,
+                //     reason: `Unknown`,
+                // })
             }
-        } catch (e) {
+        } catch (error) {
+            //TODO: use telemetry.run()
             getLogger().error(`LspController: Failed to build index of project`)
             telemetry.amazonq_indexWorkspace.emit({
                 duration: performance.now() - start,
                 result: 'Failed',
                 amazonqIndexFileCount: 0,
                 amazonqIndexFileSizeInMB: 0,
+                reason: `Error when building index. ${error instanceof Error ? error.message : error}`,
```

Comment:
`reason` is typically used as the error code/id, and `reasonDesc` is where youd put the error message

Reply:
Just saw max's earlier comment, was the change not pushed?

Reply:
Yeah it seems that this change is still making a very verbose 'reason' field which is not very useful.
```suggestion
                reason: error.code ? error.code : error.name ? error.name : 'Unkown',
                reasonDesc: `Error when building index. ${error instanceof Error ? error.message : error}`,
```